src/renderer/components/+namespaces/namespace-select-filter.tsx

```diff
@@ -72,26 +77,55 @@ export class NamespaceSelectFilter extends React.Component<SelectProps> {
   }
 
   onChange([{ value: namespace }]: SelectOption[]) {
-    if (namespace) {
+    if (NamespaceSelectFilter.isMultiSelection.get() && namespace) {
       namespaceStore.toggleContext(namespace);
+    } else if (!NamespaceSelectFilter.isMultiSelection.get() && namespace) {
+      namespaceStore.toggleSingle(namespace);
     } else {
-      namespaceStore.toggleAll(false); // "All namespaces" clicked
+      namespaceStore.toggleAll(true); // "All namespaces" clicked
     }
   }
 
+  onKeyDown = (e: React.KeyboardEvent<any>) => {
+    if (isMac && e.metaKey || isWindows && e.ctrlKey || isLinux && e.shiftKey) {
+      NamespaceSelectFilter.isMultiSelection.set(true);
+    }
+  };
+
+  onKeyUp = (e: React.KeyboardEvent<any>) => {
+    if (isMac && e.key === "Meta" || isWindows && e.key === "Ctrl" || isLinux && e.key === "Shift") {
+      NamespaceSelectFilter.isMultiSelection.set(false);
+    }
+  };
+
+  onClick = () => {
+    if (!NamespaceSelectFilter.isMultiSelection.get()) {
```

Comment:
I guess there isn't a way to learn what the keyboard state was at the time of the click? (Instead of watching keyDown and keyUp)

Reply:
Not sure.. @aleksfront @Nokel81 Wdyt?

Also, unfortunately, I could not reproduce the issue. Maybe because I have only 20 namespaces.

Reply:
Yes, we can fetch keyboard state from `onClick` event argument. There's `metaKey`, `ctrlKey` and `shiftKey` props as well.

Reply:
@vshakirova Proper type for event would be `React.MouseEvent`:
```
onClick = (event: React.MouseEvent) => {
```

Reply:
After further investigating, it seems that watching `onKeyDown` & `onKeyUp` is the best option for now. `ReactSelect` doesn't provide `onClick` event to handle this so we can't actually fetch both meta keys and selected element at once.

Reply:
@jim-docker Moved `Failed to fetch` error to separate issue https://github.com/lensapp/lens/issues/3350